### ISSUE TITLE
Crashes because of undefined function is is_plugin_active

### ISSUE THREAD

**Issue by oliverw** [btcpayserver/woocommerce-greenfield-plugin#2]
**Describe the bug**

I've installed BTCPay Server and Woocommerce using the BTCPay configurator toggling the 'woocommerce' option during setup. All worked fine until Wordpress prompted to update plugins. Since the BTCPay Woocommerce Plugin displayed the legacy advise, I installed the V2 version of plugin. After activating it the admin area is no longer available and just shows:

> There has been a critical error on this website. Please check your site admin email inbox for instructions.

Digging into the logs of the `generated_woocommerce_1` container I discovered this:

>   [Wed Feb 23 10:20:00.408285 2022] [php7:error] [pid 88] [client 172.21.0.15:46078] PHP Fatal error:  Uncaught Error: Call to undefined function is_plugin_active() in /var/www/html/wp-content/plugins/btcpay-greenfield-for-woocommerce/btcpay-greenfield-for-woocommerce.php:119\nStack trace:\n#0 /var/www/html/wp-content/plugins/btcpay-greenfield-for-woocommerce/btcpay-greenfield-for-woocommerce.php(47): BTCPayServerWCPlugin->dependenciesNotification()\n#1 /var/www/html/wp-content/plugins/btcpay-greenfield-for-woocommerce/btcpay-greenfield-for-woocommerce.php(220): BTCPayServerWCPlugin->__construct()\n#2 /var/www/html/wp-includes/class-wp-hook.php(307): init_btcpay_greenfield('')\n#3 /var/www/html/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters(NULL, Array)\n#4 /var/www/html/wp-includes/plugin.php(474): WP_Hook->do_action(Array)\n#5 /var/www/html/wp-settings.php(450): do_action('plugins_loaded')\n#6 /var/www/html/wp-config.php(96): require_once('/var/www/html/w...')\n#7 /var/www/html/wp-load.php(50): require_once('/var/www/html/w...')\n#8 /var/www/html/wp-admin/admin.php(34): require_once('/var/www/html/w...')\n# in /var/www/html/wp-content/plugins/btcpay-greenfield-for-woocommerce/btcpay-greenfield-for-woo

Editing `/var/www/html/wp-content/plugins/btcpay-greenfield-for-woocommerce/btcpay-greenfield-for-woocommerce.php` and adding this at the top of the file fixed it for now:

```
if( !function_exists('is_plugin_active') ) {
        include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
}

```
**Woocommerce System Status**

Not available since whole admin area is not available.

**WooCommerce Plugin Logs (if applicable)**

**Your BTCPay Environment (please complete the following information):**

- BTCPay Server Version:    
- Deployment Method:        Docker using Configurator
- Browser:                  Edge Chromium

**Comment by worthwhileindustries** [comment]
bumping this. Same issue WooCommerce Version 6.1.1, Wordpress Version 5.9.1 ... the previous version (V1) of the plugin worked fine.

**Comment by ndeet** [comment]
Hi thanks for reporting, sorry for the late reply, I did miss the notification that there is a new issue. 

Interesting that this pops up now. Will try to reproduce and fix and do a new release. Will keep you updated here.

**Comment by ndeet** [comment]
This is now fixed in v0.2.2 (see this [commit](https://github.com/btcpayserver/woocommerce-greenfield-plugin/commit/5f6ad99cd175d85dde8a060fb6e7cc07e7122ec0))which will be available on WP in a few minutes.

Thank you @oliverw for reporting the bug and proposing the fix. Turns out this did not hit on my end because if you have other WP plugins installed they likely also check if WC is active and already load the `plugin.php` and therefore the function is available. This check and loading of the plugin file was missing in our code and caused fatal error if no plugin else loaded it.

Let me know if the fix works for you too.
